src/d3a/models/state.py

```diff
@@ -56,17 +56,33 @@ def get_state(self) -> Dict:
 
     @abstractmethod
     def restore_state(self, state_dict: Dict):
-        pass
+        """Update the state of the device using the provided dictionary."""
 
     @abstractmethod
     def delete_past_state_values(self, time_slot):
-        pass
+        """Delete the state of the device before the given time slot."""
 
     def __str__(self):
         return self.__class__.__name__
 
 
-class ConsumptionState(StateInterface):
+class ProsumptionInterface(StateInterface, ABC):
+    """Interface with common methods/variables shared by consumption and production devices."""
+
+    def __init__(self):
+        # Actual energy consumed/produced by the device at specific market slots
+        self._real_energy: Dict[DateTime, float] = {}
+
+    def set_real_energy_kWh(self, energy_kWh: float, time_slot: DateTime) -> None:
```

Comment:
I read `-> None` a best practice? IMO it is not needed.

Reply:
I would personally treat `None` as any other type.
We are using type hints just as docs, so right now this is not that relevant (even though I like documentation). But we could decide to use a static type checker like mypy to debug our codebase. As an example, it would find these:
```python
# my_test.py
class SomeClass:
    @staticmethod
    def do_something() -> None:
        return 12
```
From console
```sh
$ mypy my_test.py
my_test.py:4: error: No return value expected
```
Note that removing `-> None:` is not the same! In that case, the static checker will just not check the return type:
```python
# my_test.py
class SomeClass:
    @staticmethod
    def do_something():
        return 12
```
```sh
$ mypy my_test.py
Success: no issues found in 1 source file
```

Reply:
ok, makes sense. I have to get used to it then. BTW, this does not apply to `__init__` ?

Reply:
I think it can apply to everything: I'm also not completely used to it but in theory you could use it everywhere. If you want, I can also add it to the `__init__`.

Reply:
IMO dont add it at all. ;-) But yeah, leave it in `set_real_energy_kWh` but omit it in `__init__`, Thanks!